packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1861,6 +1882,18 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             type: envelope.contents.type,
         };
 
+        // Only collect signal telemetry for messages sent by the current client.
+        if (message.clientId === this.clientId) {
+            if (envelope.metadata !== undefined) {
+                if (envelope.metadata.sequenceNumber !== this._perfSignalData.signalSequenceNumber) {
+                    this._perfSignalData.signalsLost++;
```

Comment:
I do not think this is correct logic. Client can send 10 signals in a row (rapidly). With this logic, we will consider all of them lost, even if no signal was lost

Reply:
My bad, on the unit tests and also while debugging a live page, perhaps because debugging slows it down,  I did not see it happening and for some reason I assumed it would not happen :)  I added a cache for it.